cloudinit/config/cc_resolv_conf.py

```diff
@@ -104,12 +109,24 @@ def handle(name, cfg, cloud, log, _args):
     if "resolv_conf" not in cfg:
         log.warning("manage_resolv_conf True but no parameters provided!")
 
-    template_fn = cloud.get_template_filename('resolv.conf')
+    distro_name = util.get_linux_distro()[0]
+    if distro_name == "photon":
+        resolv_conf = "/etc/systemd/resolved.conf"
+
+    template_fn = cloud.get_template_filename(resolv_conf)
     if not template_fn:
-        log.warning("No template found, not rendering /etc/resolv.conf")
+        log.warning("No template found, not rendering " + resolv_conf)
         return
 
-    generate_resolv_conf(template_fn=template_fn, params=cfg["resolv_conf"])
+    if distro_name == "photon":
```

Comment:
I partially understood your point here, what you are suggesting is correct but I'm not sure about achieving it via `distros/__init__.py`.

Do I need to have a function in `photon.py` which will return the `resolv_conf_fn` ? But then if user enables `- resolv_conf` in `cloud.cfg` how will `config/cc_resolv_conf.py` will consume it?

Reply:
It could be consumed with something along the lines of `cloud.distro.generate_resolv_conf()`. That would need to be defined in `distros/__init__.py` and then overridden in your distro.